ComposeTesting/README.md

```diff
@@ -0,0 +1,190 @@
+
+```kotlin
+fun createWindowLayoutInfoPublisherRule(): TestRule
+```
+
+Return WindowLayoutInfoPublisherRule which allows you to simulate the different foldingFeature by pushing through different WindowLayoutInfo values.
+
+```kotlin
+fun <A : ComponentActivity> TestRule.simulateVerticalFoldingFeature(
+    composeTestRule: AndroidComposeTestRule<ActivityScenarioRule<A>, A>,
+    center: Int = -1,
+    size: Int = 0,
+    state: FoldingFeature.State = FoldingFeature.State.HALF_OPENED
+)
+```
+
+Simulate a vertical foldingFeature in a Compose test
+
+```kotlin
+fun <A : ComponentActivity> TestRule.simulateHorizontalFoldingFeature(
+    composeTestRule: AndroidComposeTestRule<ActivityScenarioRule<A>, A>,
+    center: Int = -1,
+    size: Int = 0,
+    state: FoldingFeature.State = FoldingFeature.State.HALF_OPENED
+)
+```
+
+Simulate a horizontal foldingFeature in a Compose test
+
+```kotlin
+fun <A : ComponentActivity> TestRule.simulateFoldingFeature(
+    composeTestRule: AndroidComposeTestRule<ActivityScenarioRule<A>, A>,
+    center: Int,
+    size: Int,
+    state: FoldingFeature.State,
+    orientation: FoldingFeature.Orientation,
+) 
+```
+
+Simulate a foldingFeature with the given properties in a Compose test
+
+### Swipe Helper
+
+These functions can be used in dualscreen UI tests to simulate swipe gestures that affect
+app display. The swipes are simulated using UiDevice, and the coordinates are calculated based
+on the display width/height of the testing device.
+
+```kotlin
+fun UiDevice.spanFromStart()
+```
+
+Span app from the top/left pane
+
+```kotlin
+fun UiDevice.spanFromEnd()
+```
+
+Span app from the bottom/right pane
+
+```kotlin
+fun UiDevice.unspanToStart()
+```
+
+Unspan app to the top/left pane
+
+```kotlin
+fun UiDevice.unspanToEnd()
+```
+
+Unspan app to bottom/right pane
+
+```kotlin
+fun UiDevice.switchToStart()
+```
+
+Switch app from bottom/right pane to top/left pane
+
+```kotlin
+fun UiDevice.switchToEnd() 
+```
+
+Switch app from top/left pane to bottom/right pane
+
+```kotlin
+fun UiDevice.closeStart() 
+```
+
+Close app from top/left pane
+
+```kotlin
+fun UiDevice.closeEnd()
+```
+
+Close app from bottom/right pane
+
+
+### Device model
+
+ The DeviceModel class and related helper functions can be used in dualscreen UI tests to help
+calculate coordinates for simulated swipe gestures. Device properties are determined using
+UiDevice.
+
+```kotlin
+fun UiDevice.isSurfaceDuo(): Boolean
+```
+
+Checks whether a device is a Surface Duo model
+
+```kotlin
+fun UiDevice.getFoldSize(): Int
+```
+
+Returns a pixel value of the hinge/fold size of a foldable or dual-screen device
+
+```kotlin
+fun UiDevice.getDeviceModel(): DeviceModel
+```
+
+Returns the model of a device based on display width and height
+
```

Comment:
I would say maybe these methods don't need to be in the readme? Up to you!

I forgot to use it in `SwipeHelperTest.kt`, but basically I think the most useful method is `isSurfaceDuo()`, because they can use that to check whether or not the swipe functions can be used - ex:
```kotlin
@Test
fun spanDevice() {
    if (device.isSurfaceDuo())
        device.spanFromStart()
}
```

Reply:
Oh also, if you also agree, I think it would be good to add the `StringHelper`, `ZoomHelper`, and `ScreenshotsComparator` methods to the end here - they don't need anything more than the header comments, but it's probably good to show

Reply:
I would skip `ScreenshotsComparator` since it is from Google's codelab.

Reply:
Sounds good, the updates look great!